synapse/federation/federation_server.py

```diff
@@ -107,23 +110,25 @@ def __init__(self, hs):
 
         return 200, res
 
-    async def on_incoming_transaction(self, origin, transaction_data):
+    async def on_incoming_transaction(
+        self, origin: str, transaction_data: Dict[str, Any]
```

Comment:
typically I'd suggest using `JsonDict` for this sort of thing.

Reply:
Nice, didn't know of that one. 👍